Source/WebCore/dom/MouseRelatedEvent.cpp

```diff
@@ -86,13 +95,13 @@ static inline bool isMoveEventType(const AtomString& eventType)
         || eventType == eventNames.touchmoveEvent;
 }
 
-void MouseRelatedEvent::init(bool isSimulated, const IntPoint& windowLocation)
+void MouseRelatedEvent::init(bool isSimulated, const DoublePoint& windowLocation)
 {
     if (!isSimulated) {
         if (RefPtr frameView = frameViewFromWindowProxy(view())) {
-            FloatPoint absolutePoint = frameView->windowToContents(windowLocation);
-            FloatPoint documentPoint = frameView->absoluteToDocumentPoint(absolutePoint);
-            m_pageLocation = flooredLayoutPoint(documentPoint);
+            DoublePoint absolutePoint = frameView->windowToContents(windowLocation);
+            DoublePoint documentPoint = frameView->absoluteToDocumentPoint(absolutePoint);
+            m_pageLocation = DoublePoint(documentPoint);
```

Comment:
You already have a DoublePoint so no need to create one again
```suggestion
            DoublePoint documentPoint = frameView->absoluteToDocumentPoint(absolutePoint);
            m_pageLocation = WTFMove(documentPoint);
```